mirascope/core/anthropic/stream.py

```diff
@@ -34,9 +40,67 @@ def cost(self) -> float | None:
     def _construct_message_param(
         self, tool_calls: list[ToolUseBlock] | None = None, content: str | None = None
     ) -> MessageParam:
-        """Returns the tool message parameters for tool call results."""
+        """Constructs the message parameter for the assistant."""
+        message_param_tool_calls = []
+        if tool_calls:
+            message_param_tool_calls: list[ToolUseBlockParam] = [
+                tool_call.model_dump()
+                for tool_call in tool_calls  # type: ignore
+            ]
         return {
             "role": "assistant",
-            "content": ([TextBlock(text=content, type="text")] if content else [])
-            + (tool_calls or []),
+            "content": ([TextBlockParam(text=content, type="text")] if content else [])
+            + message_param_tool_calls,
         }
+
+    def construct_call_response(self) -> AnthropicCallResponse:
+        if self.message_param is None:
+            raise ValueError(  # pragma: no cover
+                "No stream response, check if the stream has been consumed."
+            )
+        usage = Usage(input_tokens=0, output_tokens=0)
+        if self.input_tokens:
+            usage.input_tokens = int(self.input_tokens)
+        if self.output_tokens:
+            usage.output_tokens = int(self.output_tokens)
+
+        message_param: list[ContentBlock] = []
+
+        if isinstance(self.message_param["content"], str):
+            message_param.append(
+                TextBlock(text=self.message_param["content"], type="text")
+            )
+        else:
+            for content in self.message_param["content"]:
+                content_type = (
+                    content.type if isinstance(content, BaseModel) else content["type"]
+                )
+
+                if content_type == "text":
+                    message_param.append(TextBlock.model_validate(content))
+                elif content_type == "tool_use":
+                    message_param.append(ToolUseBlock.model_validate(content))
+        completion = Message(
+            id=self.id if self.id else "",
+            content=message_param,
+            model=self.model,
+            role="assistant",
+            stop_reason=self.finish_reasons[0] if self.finish_reasons else None,  # type: ignore
```

Comment:
why is there a type ignore here? If it's because of something with the base type, then we should be using generics to properly type this.

Reply:
should we pass in finish reason as a generic type? I feel like if thats the case I should do that for all inputs. Ideally I would be able to declare a property but I wasn't sure how to go about that

Reply:
done